### PR TITLE
Implement bubble sort

### PR DESCRIPTION
**Pull request by edemagbenyo**
Bubble sort implementation in ruby using:
while loop
times method 
block

### CODE REVIEW

File: index.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  end
+  arr
+end
+
+bubble_sort([9, 7, 6, 5, 4, 3, 2, 1])
+
+def bubble_sort_by(arr)
+  k = arr.size
+  count = 0
+  k.times do |i|
```
Comment (samgaco) [inline index.rb:24]: - [ ] **bubble_sort_by** should raise an error if no block is given

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @edemagbenyo and @kilonzi :high_brightness: 

Everything looks perfect and love the readme file :100: Just one thing left to correct:

- [ ]  bubble_sort_by should raise an error if no block is given

Please work on that and submit back for another code review. Regards! :muscle: 

----------------------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: . However the project didn't follow the GitFlow guidelines. check below :arrow_down: 

- Since you are going to use this repo later in the program, you are supposed to create an intermediary branch "development" to which you are supposed to merge(i.e not master branch).
- create a development branch, set target branch of this pull request to be `development`

### That's it :relaxed: When finished, submit another code review request :pray: 

**Ahmed Mahfoudh**

File: index.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    n = 0
+    while n < (arr.size - 1 - i)
+      count += 1
+      raise 'No block given. Please supply a block.' unless block_given?
```
Comment (maelfosso) [inline index.rb]: It's not the good place.
Putting it here mean that you're going to test it every times in the while loop.
Think about put it before the starting the while loop or after ```def bubble_sort_by(arr)``` ?

File: index.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    while n < (arr.size - 1 - i)
+      count += 1
+      arr[n], arr[n + 1] = arr[n + 1], arr[n] if arr[n] > arr[n + 1]
+      n += 1
+    end
+  end
+  arr
+end
+
+bubble_sort([9, 7, 6, 5, 4, 3, 2, 1])
```
Comment (maelfosso) [inline index.rb:17]: Not needed

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

Thank you for the improvement. The **Git Flow** process is OK now.
Just one bug to fix. Even if it's correctly running, the `raise error` it's not at the good position. Read the comment below, please.

Kindly make the changes and submit a new code review request.
Thanks.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by maelfosso — APPROVED** [review]
# You did it :+1: The project is APPROVED :clap: 

You can merge it.
Good luck with the next step

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**